### PR TITLE
Fix module loading from S3

### LINKED ISSUE #828: xcube serve Datasets configuration to use s3 resources
**Describe the bug**
When configuring Datasets via xcube serve not using the DataStore framework, s3 resources cause errors. 

**To Reproduce**
Use the configuration: 

```yaml

base_dir: s3://xcube-viewer-app/ws/dev

Viewer:
  Configuration:
    Path: viewer

Datasets:
  - Identifier: app-l2c-cyanoalert-olci-balt-v2
    Path: water-services/Sweden/app-l2c-cyanoalert-olci-balt-Sweden-v2.zarr
    Title: Sverige app v2 - limited temporal length
    FileSystem: s3
    Endpoint: "https://s3.eu-central-1.amazonaws.com"
    Anonymous: false
    Style: default

  - Identifier: c-app-l2c-cyanoalert-olci-balt-v2-1D
    Path: api/augmentation/resample_in_time.py
    Title: Sverige app v2 1D- limited temporal length
    Style: default
    FileSystem: memory
    Function: compute_dataset
    InputDatasets: [app-l2c-cyanoalert-olci-balt-v2]
    InputParameters:
      period: 1D
```

Traceback:
```
[I 2023-03-03 12:06:50,791 xcube] Opened dataset 'app-l2c-cyanoalert-olci-balt-v2' from data store 's3_1': took 326.93ms
[W 2023-03-03 12:06:50,792 xcube] Skipping dataset app-l2c-cyanoalert-olci-balt-v2: Parameter validation failed:
Invalid bucket name "s3:": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
Traceback (most recent call last):
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/controllers.py", line 140, in get_datasets
    get_dataset(ctx, ds_id,
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/controllers.py", line 182, in get_dataset
    ml_ds = ctx.get_ml_dataset(ds_id)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 183, in get_ml_dataset
    ml_dataset, _ = self._get_dataset_entry(ds_id)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 602, in _get_dataset_entry
    self._set_dataset_entry(self._create_dataset_entry(ds_id))
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 615, in _create_dataset_entry
    ml_dataset = self._open_ml_dataset(dataset_config)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/webapi/datasets/context.py", line 639, in _open_ml_dataset
    dataset = data_store.open_data(data_id, **open_params)
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/core/store/fs/store.py", line 333, in open_data
    return opener.open_data(fs_path,
  File "/opt/conda/lib/python3.9/site-packages/xcube-1.0.0.dev2-py3.9.egg/xcube/core/store/fs/impl/dataset.py", line 190, in open_data
    fs.exists(f'{data_id}/.zmetadata'))
  File "/opt/conda/lib/python3.9/site-packages/fsspec/asyn.py", line 114, in wrapper
    return sync(self.loop, func, *args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/fsspec/asyn.py", line 99, in sync
    raise return_result
  File "/opt/conda/lib/python3.9/site-packages/fsspec/asyn.py", line 54, in _runner
    result[0] = await coro
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 974, in _exists
    await self._info(path, bucket, key, version_id=version_id)
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 1238, in _info
    out = await self._call_s3(
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 339, in _call_s3
    return await _error_wrapper(
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 139, in _error_wrapper
    raise err
  File "/opt/conda/lib/python3.9/site-packages/s3fs/core.py", line 112, in _error_wrapper
    return await func(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/client.py", line 321, in _make_api_call
    request_dict = await self._convert_to_request_dict(
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/client.py", line 383, in _convert_to_request_dict
    api_params = await self._emit_api_params(
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/client.py", line 416, in _emit_api_params
    await self.meta.events.emit(
  File "/opt/conda/lib/python3.9/site-packages/aiobotocore/hooks.py", line 66, in _emit
    response = await resolve_awaitable(handler(**kwargs))
  File "/opt/conda/lib/python3.9/site-packages/botocore/handlers.py", line 275, in validate_bucket_name
    raise ParamValidationError(report=error_msg)
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid bucket name "s3:": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:(s3|s3-object-lambda):[a-z\-0-9]*:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-.]{1,63}$|^arn:(aws).*:s3-outposts:[a-z\-0-9]+:[0-9]{12}:outpost[/:][a-zA-Z0-9\-]{1,63}[/:]accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
```

**Expected behavior**
I would expect that the Datasets will be able to use the s3 resources the same way the DataStore configuration is.

### PR DESCRIPTION
**Pull request by forman**
Fixed recurring issue where xcube server was unable to locate Python code downloaded from S3 when configuring dynamically computed datasets (configuration `FileSystem: memory`) or augmenting existing datasets by dynamically computed variables (configuration `Augmentation`). 

Closes #828

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Changes documented in `CHANGES.md`
* [ ] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

File: test/core/byoa/test_fileset.py [hunk 1]
```diff
@@ -1,20 +1,20 @@
 import os
 import os.path
 import os.path
+import os.path
```
Comment (pont-us) [inline test/core/byoa/test_fileset.py:4]:
```suggestion
import os.path
```

I think once is enough!

**Review by pont-us — APPROVED** [review]
Looks good!

I've made one suggested change (remove duplicated import lines in the test) but approving since it's not important.

File: test/core/byoa/test_fileset.py (hunk 1 above)
Comment (forman) [reply test/core/byoa/test_fileset.py:4]: :)

**Review by AliceBalfanz — APPROVED** [review]
Tested locally, and works nicely. Looking forward to test in the cloud!

**Pull request merged**